iothub_client/samples/pnp/pnp_temperature_controller/pnp_temperature_controller.c

```diff
@@ -289,21 +306,122 @@ static void PnP_TempControlComponent_ReportSerialNumber_Property(IOTHUB_DEVICE_C
 }
 
 
+//
+// GetConnectionStringFromEnvironment retrieves the connection string based on environment variable
+//
+static bool GetConnectionStringFromEnvironment()
+{
+    bool result;
+
+    if ((g_pnpDeviceConfiguration.u.connectionString = getenv(g_connectionStringEnvironmentVariable)) == NULL)
+    {
+        LogError("Cannot read environment variable=%s", g_connectionStringEnvironmentVariable);
+        result = false;
+    }
+    else
+    {
+        result = true;    
+    }
+
+    return result;
+}
+
+//
+// GetDpsFromEnvironment retrieves DPS configuration for a symmetric key based connection
+// from environment variables
+//
+static bool GetDpsFromEnvironment()
+{
+#ifndef USE_PROV_MODULE
+    // Explain to user misconfiguration.  The "run_e2e_tests" must be set to OFF because otherwise
+    // the e2e's test HSM layer and symmetric key logic will conflict.
+    LogError("DPS based authentication was requested via environment variables, but DPS is not enabled.");
+    LogError("DPS is an optional component of the Azure IoT C SDK.  It is enabled with symmetric keys at cmake time by");
+    LogError("passing <-Duse_prov_client=ON -Dhsm_type_symm_key=ON -Drun_e2e_tests=OFF> to cmake's command line");
+    return false;
+#else
+    bool result;
+
+    if ((g_pnpDeviceConfiguration.u.dpsConfiguration.idScope = getenv(g_dpsIdScopeEnvironmentVariable)) == NULL)
+    {
+        LogError("Cannot read environment variable=%s", g_dpsIdScopeEnvironmentVariable);
+        result = false;
+    }
+    else if ((g_pnpDeviceConfiguration.u.dpsConfiguration.registrationId = getenv(g_dpsRegistrationIdEnvironmentVariable)) == NULL)
+    {
+        LogError("Cannot read environment variable=%s", g_dpsRegistrationIdEnvironmentVariable);
+        result = false;
+    }
+    else if ((g_pnpDeviceConfiguration.u.dpsConfiguration.deviceKey = getenv(g_dpsDeviceKeyEnvironmentVariable)) == NULL)
+    {
+        LogError("Cannot read environment variable=%s", g_dpsDeviceKeyEnvironmentVariable);
+        result = false;
+    }
+    else
+    {
+        result = true;    
+    }
+
+    return result;
+#endif
```

Comment:
```suggestion
#endif // USE_PROV_MODULE
```

Reply:
Taking (putting in code directly instead of via commit suggestion).